birdhouse/README.rst

```diff
@@ -403,6 +403,79 @@ In order to set up the development environment, run the following commands:
     # Activate the conda environment
     conda activate birdhouse-dev
 
+
+Development Conventions
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The aim should be to **avoid** backward **incompatible** changes to decrease manual effort between each
+update. When the effort is low, users will be more likey to stay up to date and not "fear" updates.
+
+We should also aim to be **user-friendly** by being consistent with the way we grow the platform and by
+requiring as little setup/steps (sensible default values) as possible when spin up a fresh new stack
+for onboarding new users.
+
+Adding a new config var to ``env.local``
+========================================
+* **Default value** should be in a corresponding ``default.env`` so it is easy for the user to find them.
+  The default value should not be burried in the code.  The commented out value in ``env.local.example``
+  is only an example and can not count as a default value.
+
+* **Documentation** for the new var can be both in ``default.env`` and ``env.local.example`` to be
+  most user-friendly. If you do not wish to duplicate the info because it is big, you can put in one of
```

Comment:
I think we need to choose which one we prefer. I would say that we should always prefer `env.local.example` for variables that are use facing and `default.env` for the internal-only variables.